lib/charms/kubernetes_charm_libraries/v0/kubernetes_hugepages_volumes_patch.py

```diff
@@ -0,0 +1,624 @@
+        container_name: str,
+        requested_volumemounts: Iterable[VolumeMount],
+    ) -> bool:
+        """Returns whether container spec contains the given volumemounts.
+
+        Args:
+            containers: Iterable of Containers
+            container_name: Container name
+            requested_volumemounts: Iterable of volume mounts that the container shall contain
+
+        Returns:
+            bool: whether container spec contains the given volumemounts.
+        """
+        container = self._get_container(container_name=container_name, containers=containers)
+        return all(
+            [
+                requested_volumemount in container.volumeMounts
+                for requested_volumemount in requested_volumemounts
+            ]
+        )
+
+    def _pod_resources_are_set(
+        self,
+        containers: Iterable[Container],
+        container_name: str,
+        requested_resources: ResourceRequirements,
+    ) -> bool:
+        """Returns whether container spec contains the expected resources requests and limits.
+
+        Args:
+            containers: Iterable of Containers
+            container_name: Container name
+            requested_resources: resource requirements
+
+        Returns:
+            bool
+        """
+        container = self._get_container(container_name=container_name, containers=containers)
+        if requested_resources.limits:
+            for limit, value in requested_resources.limits.items():
+                if not container.resources.limits:
+                    return False
+                if container.resources.limits.get(limit) != value:
+                    return False
+        if requested_resources.requests:
+            for request, value in requested_resources.requests.items():
+                if not container.resources.requests:
+                    return False
+                if container.resources.requests.get(request) != value:
+                    return False
+        return True
+
+    def replace_statefulset(
+        self,
+        statefulset_name: str,
+        requested_volumes: Iterable[Volume],
+        requested_volumemounts: Iterable[VolumeMount],
+        requested_resources: ResourceRequirements,
+        container_name: str,
+    ) -> None:
+        """Updates a StatefulSet and a container in its spec.
+
+        Args:
+            statefulset_name: Statefulset name
+            requested_volumes: Iterable of new volumes to be set in the StatefulSet
+            requested_volumemounts: Iterable of new volume mounts to be set in the given container
+            requested_resources: new resource requirements to be set in the given container
+            container_name: Container name
+        """
+        if not requested_volumes:
+            logger.warning("No requested volumes were provided")
+            return
+        if not requested_volumemounts:
+            logger.warning("No requested volume mounts were provided")
+            return
```

Comment:
I'm not sure about this logic here. The name of the method doesn't suggest any relation to Volumes or VolumeMounts. I think it would be cleaner if this method only had logic for replacing the StatefulSet.